flaml/integrations/oai/completion.py

```diff
@@ -385,23 +454,54 @@ def tune(
             metric (str): The metric to optimize.
             mode (str): The optimization mode, "min" or "max.
             eval_func (Callable): The evaluation function for responses.
+                The function should take a list of responses and a data point as input,
+                and return a dict of metrics. For example,
+
+            ```python
+            def eval_func(responses, **data):
+                solution = data["solution"]
+                success_list = []
+                n = len(responses)
+                for i in range(n):
+                    response = responses[i]
+                    succeed = is_equiv_chain_of_thought(response, solution)
+                    success_list.append(succeed)
+                return {
+                    "expected_success": 1 - pow(1 - sum(success_list) / n, n),
+                    "success": any(s for s in success_list),
+                }
+            ```
+
             log_file_name (str, optional): The log file.
             inference_budget (float, optional): The inference budget.
             optimization_budget (float, optional): The optimization budget.
             num_samples (int, optional): The number of samples to evaluate.
+                -1 means no hard restriction in the number of trials
+                and the actual number is decided by optimization_budget. Defaults to 1.
             **config (dict): The search space to update over the default search.
-                For prompt, please provide a string or a list of strings.
+                For prompt, please provide a string/Callable or a list of strings/Callables.
+                    - If prompt is provided for chat models, it will be converted to messages under role "user".
+                    - Do not provide both prompt and messages for chat models, but provide either of them.
+                    - A string `prompt` template will be used to generate a prompt for each data instance
```

Comment:
should we stick to OpenAI, i.e., prompt for completion, and messages for chat? 

for chat, are prompt/messages used interchangeably?

Reply:
The reason is to provide a single API that can be used to tune both chat models and non-chat models. For chat models, users can provide either prompt or messages. But for non-chat models, only prompt. So they are not used interchangeably.

Reply:
Right, I mean for OpenAI, chat API only has messages, completion API only has prompt

For FLAML, chat API has both messages and prompt, completion API has prompt. Would that be confusing to users?

Reply:
FLAML offers a superset, so I think it's OK. The other way around would be problematic. If a user doesn't need to use the extra capabilities, it's fine.